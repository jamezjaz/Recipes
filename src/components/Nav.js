import React from 'react';
import { Link } from 'react-router-dom';
import NavStyles from '../styles/Nav.module.css';
import logo from '../assets/logo.png';

const Nav = () => (
  <nav className={NavStyles.nav}>
    <h2
      className="my-logo"
      style={{ color: 'white' }}
    >
      <Link to="/">
        <img src={logo} alt={logo} />
      </Link>
      Catalogue Of Recipes
    </h2>
    <ul className={NavStyles.navLinks}>
      <Link className={NavStyles.lists} to="/">
        <li>Recipes</li>
      </Link>
      <Link className={NavStyles.lists} to="/about">
        <li>About Us</li>
      </Link>
    </ul>
  </nav>
);

export default Nav;
