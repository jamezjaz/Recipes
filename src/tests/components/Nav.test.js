import React from 'react';
import { cleanup, screen, render } from '@testing-library/react';
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter as Router, Link } from 'react-router-dom';
import Nav from '../../components/Nav';

Enzyme.configure({ adapter: new Adapter() });

describe('<Nav />', () => {
  afterEach(cleanup);
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Nav />);
  });

  test('renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
            <Nav />
        </Router>,
      )
      .toJSON();
      expect(tree).toMatchSnapshot();
  });

  test('contains Catalogue Of Recipes header', () => {
    render(
      <Router>
        <Nav />
      </Router>,
    );
    expect(screen.getByText('Catalogue Of Recipes')).toBeInTheDocument();
  });

  test('it contains li element', () => {
    const li = <li>Recipes</li>;
    expect(wrapper.containsMatchingElement(li)).toBe(true);
  });

  test('h2 element has a text', () => {
    const h2 = wrapper.find('.my-logo');
    expect(h2.text()).toEqual('Catalogue Of Recipes');
  });

  test('it contains a Link', () => {
    const link = <Link><li>About Us</li></Link>;
    expect(wrapper.containsMatchingElement(link)).toBe(true);
  });
});

