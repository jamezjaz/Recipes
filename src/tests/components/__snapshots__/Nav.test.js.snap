// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> renders correctly 1`] = `
<nav
  className="nav"
>
  <h2
    className="my-logo"
    style={
      Object {
        "color": "white",
      }
    }
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="logo.png"
        src="logo.png"
      />
    </a>
    Catalogue Of Recipes
  </h2>
  <ul
    className="navLinks"
  >
    <a
      className="lists"
      href="/"
      onClick={[Function]}
    >
      <li>
        Recipes
      </li>
    </a>
    <a
      className="lists"
      href="/about"
      onClick={[Function]}
    >
      <li>
        About Us
      </li>
    </a>
  </ul>
</nav>
`;
